File: reports/20260826-143732/results/key-vault/data-plane/python/crud/key-vault-dp-python-crud/python-azure-tools/baseline/generated-code/key_vault_secret_crud.py
"""Perform create, read, update, and delete operations on an Azure Key Vault secret."""

import os
import sys

from azure.core.exceptions import AzureError, ResourceNotFoundError
from azure.identity import DefaultAzureCredential
from azure.keyvault.secrets import SecretClient


SECRET_NAME = os.getenv("AZURE_KEY_VAULT_SECRET_NAME", "crud-demo-secret")
INITIAL_VALUE = "initial-value"
UPDATED_VALUE = "updated-value"


def require_vault_url() -> str:
    vault_url = os.getenv("AZURE_KEY_VAULT_URL")
    if not vault_url:
        raise ValueError(
            "AZURE_KEY_VAULT_URL is required (for example, "
            "https://your-vault-name.vault.azure.net)."
        )
    return vault_url


def run_crud_operations() -> None:
    credential = DefaultAzureCredential()
    client = SecretClient(vault_url=require_vault_url(), credential=credential)

    try:
        created_secret = client.set_secret(SECRET_NAME, INITIAL_VALUE)
        print(f"Created secret '{created_secret.name}'.")

        read_secret = client.get_secret(SECRET_NAME)
        print(f"Read secret '{read_secret.name}' with value '{read_secret.value}'.")

        updated_secret = client.set_secret(SECRET_NAME, UPDATED_VALUE)
        print(
            f"Updated secret '{updated_secret.name}' "
            f"to value '{updated_secret.value}'."
        )

        delete_poller = client.begin_delete_secret(SECRET_NAME)
        delete_poller.wait()
        print(f"Deleted secret '{SECRET_NAME}'.")

        # Waiting for deletion ensures the secret is available in the deleted
        # secrets collection before the purge request is sent.
        client.get_deleted_secret(SECRET_NAME)
        client.purge_deleted_secret(SECRET_NAME)
        print(f"Purged secret '{SECRET_NAME}'.")
    finally:
        client.close()
        credential.close()


def main() -> int:
    try:
        run_crud_operations()
        return 0
    except ValueError as error:
        print(f"Configuration error: {error}", file=sys.stderr)
    except ResourceNotFoundError as error:
        print(f"Secret was not found during the operation: {error}", file=sys.stderr)
    except AzureError as error:
        print(f"Azure Key Vault operation failed: {error}", file=sys.stderr)
    except KeyboardInterrupt:
        print("Operation cancelled.", file=sys.stderr)
    return 1


if __name__ == "__main__":
    raise SystemExit(main())

File: rerun-reports/lane-a/021/20260828-012928/results/key-vault/data-plane/java/crud/key-vault-dp-java-crud/java-azure-skills/azure-skill-mcp-microsoft-skill/generated-code/src/main/java/com/example/KeyVaultSecretCrud.java
package com.example;

import com.azure.core.exception.ClientAuthenticationException;
import com.azure.core.exception.HttpResponseException;
import com.azure.core.exception.ResourceNotFoundException;
import com.azure.core.util.polling.SyncPoller;
import com.azure.identity.DefaultAzureCredentialBuilder;
import com.azure.security.keyvault.secrets.SecretClient;
import com.azure.security.keyvault.secrets.SecretClientBuilder;
import com.azure.security.keyvault.secrets.models.DeletedSecret;
import com.azure.security.keyvault.secrets.models.KeyVaultSecret;

public final class KeyVaultSecretCrud {
    private static final String SECRET_NAME = "my-secret";
    private static final String INITIAL_VALUE = "my-secret-value";
    private static final String UPDATED_VALUE = "updated-value";

    private KeyVaultSecretCrud() {
    }

    public static void main(String[] args) {
        String vaultUrl = System.getenv("AZURE_KEYVAULT_URL");
        if (vaultUrl == null || vaultUrl.isBlank()) {
            System.err.println(
                "AZURE_KEYVAULT_URL must be set, for example: https://<vault-name>.vault.azure.net");
            System.exit(2);
        }

        try {
            SecretClient secretClient = new SecretClientBuilder()
                .vaultUrl(vaultUrl)
                .credential(new DefaultAzureCredentialBuilder().build())
                .buildClient();

            KeyVaultSecret created = secretClient.setSecret(SECRET_NAME, INITIAL_VALUE);
            System.out.printf("Created secret \"%s\" (version %s).%n",
                created.getName(), created.getProperties().getVersion());

            KeyVaultSecret read = secretClient.getSecret(SECRET_NAME);
            System.out.printf("Read secret value: %s%n", read.getValue());

            // Secret values are immutable; setting the same name creates a new version.
            KeyVaultSecret updated = secretClient.setSecret(SECRET_NAME, UPDATED_VALUE);
            System.out.printf("Updated secret \"%s\" to version %s.%n",
                updated.getName(), updated.getProperties().getVersion());

            SyncPoller<DeletedSecret, Void> deletePoller =
                secretClient.beginDeleteSecret(SECRET_NAME);
            deletePoller.waitForCompletion();
            System.out.printf("Deleted secret \"%s\".%n", SECRET_NAME);

            secretClient.purgeDeletedSecret(SECRET_NAME);
            System.out.printf("Purged secret \"%s\" permanently.%n", SECRET_NAME);
        } catch (ClientAuthenticationException e) {
            System.err.println("Authentication failed. Check the credentials available to "
                + "DefaultAzureCredential: " + e.getMessage());
            System.exit(1);
        } catch (ResourceNotFoundException e) {
            System.err.println("The requested vault or secret was not found: " + e.getMessage());
            System.exit(1);
        } catch (HttpResponseException e) {
            int statusCode = e.getResponse() == null ? -1 : e.getResponse().getStatusCode();
            System.err.printf("Azure Key Vault request failed (HTTP %d): %s%n",
                statusCode, e.getMessage());
            if (statusCode == 403) {
                System.err.println("Ensure the identity has secret get, set, delete, and purge permissions. "
                    + "Purge also fails when purge protection is enabled.");
            } else if (statusCode == 429) {
                System.err.println("The vault is throttling requests; retry after the server's delay.");
            }
            System.exit(1);
        } catch (IllegalArgumentException e) {
            System.err.println("Invalid Key Vault configuration: " + e.getMessage());
            System.exit(2);
        }
    }
}
